paper-server/patches/sources/net/minecraft/world/entity/LivingEntity.java.patch

```diff
@@ -1118,7 +1122,10 @@
 +
 +            // Apply damage to armor
 +            if (!damageSource.is(DamageTypeTags.BYPASSES_ARMOR)) {
-+                float armorDamage = (float) (event.getDamage() + event.getDamage(DamageModifier.BLOCKING) + event.getDamage(DamageModifier.HARD_HAT));
++                float armorDamage = (float) event.getDamage();
```

Comment:
This computation is flawed anyway since it doesn't take in account the recent damage modifier added by Spigot (freezing) accounted in Vanilla. So the armor generally take less damage in some environment.

For example with a datapack that clear the #bypasses_armor damage type tag and add the player to the #freeze_hurts_extra_types entity type tag.